File: ReservedYouFrontEnd/src/app/models/Schedule.ts
export class Schedule {
    ScheduleId:string;
    BusServiceId:string;
    RouteId:string;
    RegistrationNumber:string;
    ScheduleDate:Date;
    DepartureTime:string;
    ArrivalTime:string;
    BusFare:string;
    Approved:boolean;

    constructor(sId:string,bsId:string,rId:string,rNum:string,sDate:Date,dTime:string,aTime:string,bFare:string,apr:boolean){
        this.ScheduleId = sId;
        this.BusServiceId = bsId;
        this.RouteId = rId;
        this.RegistrationNumber = rNum;
        this.ScheduleDate = new Date(sDate);
        this.DepartureTime = dTime;
        this.ArrivalTime = aTime;
        this.BusFare = bFare;
        this.Approved = apr;
    }
    
}

File: Domain/Schedule.cs
namespace Model
{
    public class Schedule
    {
        public Guid? ScheduleId { get; set; }
        public Guid? BusServiceId { get; set; }
        public Guid? RouteId { get; set; }

        // Change this to required primary key 
        public string? RegistrationNumber { get; set; }

        public DateTime ScheduleDate { get; set; }  

        public DateTime DepartureTime { get; set; }
        public DateTime ArrivalTime { get; set; }

        public int BusFare { get; set; }

        // Only Admin can Update this field
        public bool? Approved { get; set; }
        public List<Booking>? Bookings { get; set; }
        public BusService? BusService { get; set; }

        public Routes? Route { get; set; }

        public Bus? Bus { get; set; }


    }
}